server/src/main/java/com/datastrato/gravitino/server/web/rest/CatalogOperations.java

```diff
@@ -76,12 +83,16 @@ public Response createCatalog(
             request.validate();
             NameIdentifier ident = NameIdentifier.ofCatalog(metalake, request.getName());
             Catalog catalog =
-                manager.createCatalog(
+                TreeLockUtils.doWithTreeLock(
                     ident,
```

Comment:
I think you need to lock the metalake with write lock for creating, right? Otherwise, the list operation will not be consistent.

Reply:
Think twice, I think it is ok to add write lock on the catalog level.

Reply:
Still, I think you need to add the write lock in metalake level, otherwise, the list operation will be inconsistent (if I suppose the list operation get the metalake read lock like I mentioned above).